pygmo/core.cpp

```diff
@@ -885,8 +892,80 @@ BOOST_PYTHON_MODULE(core)
+                 return hv.exclusive(p_idx, pygmo::to_vd(r_point), *hv_algo);
+             },
+             "Computes the hypervolume using the provided hypervolume algorithm.",
+             (bp::arg("idx"), bp::arg("ref_point"), bp::arg("hv_algo")))
+        .def("least_contributor",
+             +[](hypervolume &hv, const bp::object &r_point) { return hv.least_contributor(pygmo::to_vd(r_point)); },
+             "Computes the least_contributor.", (bp::arg("ref_point")))
+        .def("least_contributor",
+             +[](hypervolume &hv, const bp::object &r_point, boost::shared_ptr<hv_algorithm> hv_algo) {
+                 return hv.least_contributor(pygmo::to_vd(r_point), *hv_algo);
+             },
+             "Computes the least_contributor using the provided hypervolume algorithm.",
+             (bp::arg("ref_point"), bp::arg("hv_algo")))
+        .def("greatest_contributor",
+             +[](hypervolume &hv, const bp::object &r_point) { return hv.greatest_contributor(pygmo::to_vd(r_point)); },
+             "Computes the greatest_contributor.", (bp::arg("ref_point")))
+        .def("greatest_contributor",
+             +[](hypervolume &hv, const bp::object &r_point, boost::shared_ptr<hv_algorithm> hv_algo) {
+                 return hv.greatest_contributor(pygmo::to_vd(r_point), *hv_algo);
+             },
+             "Computes the greatest_contributor using the provided hypervolume algorithm.",
+             (bp::arg("ref_point"), bp::arg("hv_algo")))
+        .def("contributions",
+             +[](hypervolume &hv, const bp::object &r_point) { return hv.contributions(pygmo::to_vd(r_point)); },
+             "Get the contributions to the hypervolume by each point.", (bp::arg("ref_point")))
+        .def("contributions",
+             +[](hypervolume &hv, const bp::object &r_point, boost::shared_ptr<hv_algorithm> hv_algo) {
+                 return hv.contributions(pygmo::to_vd(r_point), *hv_algo);
+             },
+             "Get the contributions to the hypervolume by each point using provided hypervolume algorithm..",
+             (bp::arg("ref_point"), bp::arg("hv_algo")))
+        .def("set_copy_points", &hypervolume::set_copy_points)
+        .def("get_copy_points", &hypervolume::get_copy_points)
+        .def("get_points", +[](hypervolume &hv) { return pygmo::vv_to_a(hv.get_points()); })
+        .def("set_verify", &hypervolume::set_verify)
+        .def("get_verify", &hypervolume::get_verify);
+    // Hypervolume algorithms
+    bp::class_<hv_algorithm, boost::noncopyable>("_hv_algorithm", bp::no_init).def("get_name", &hv_algorithm::get_name);
+    bp::class_<hvwfg, bp::bases<hv_algorithm>>("wfg", "WFG algorithm.")
+        .def(bp::init<const unsigned int>((bp::arg("stop_dimension") = 2)));
+    bp::class_<bf_approx, bp::bases<hv_algorithm>>("bf_approx", "Bringmann-Friedrich approximated algorithm.")
+        .def(bp::init<const bool, const unsigned int, const double, const double, const double, const double,
```

Comment:
Yes, I am fighting to put the code to a decent standard, but there are so many of these things that fighting all is impossible.